### PR TITLE
[WIP] Agent integration

### PR DESCRIPTION
**Pull request by avishayt**
Just to take an initial look - needs rebase and tests.

### CODE REVIEW

File: cmd/cluster/none/create.go [hunk 1]
```diff
@@ -93,6 +95,7 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 
 	exampleOptions.None = &apifixtures.ExampleNoneOptions{
 		APIServerAddress: opts.NonePlatform.APIServerAddress,
+		AgentBased:       opts.NonePlatform.AgentBased,
```
Comment (hardys) [inline cmd/cluster/none/create.go:98]: Maybe a nit but we're handling this new agent option inconsistently - elsewhere it's defined as a new platform, but here it's exposed as an option to the None platform.

Right now, that's probably not a big issue, but we've discussed some ideas around combining the agent provider with infrastructure automation (e.g so the agent image can be booted automatically on metal, vmware, etc).

We haven't resolved if/how to enable that yet, but I'm wondering if treating agent as a platform option consistently throughout the code would perhaps reduce friction in future if we decide to persue automating starting the agent image?

Comment (avishayt) [reply cmd/cluster/none/create.go:98]: Yes, I have an action item to adjust that.

**Comment by enxebre** [comment]
/hold
Thanks! Let's get this in first https://github.com/openshift/hypershift/pull/719 before introducing any platform extension work.

**Pull request closed without merging**